tests/test_optimize.py

```diff
@@ -0,0 +1,78 @@
+import numpy as np
+from unittest import TestCase
+
+from aspire.utils.matlab_compat import rand
+from aspire.utils.optimize import conj_grad
+
+
+def identity(input_x):
+        return input_x
+
+
+def apply(A, x):
+    p = np.size(A[0], 0)
+    x = x.reshape(p, p)
+    y = np.zeros_like(x)
+    for k in range(0, len(A)):
+            y = y + A[k] @ x @ A[k].T
+    y = y.reshape(p ** 2,)
+    return y
+
+
+class OptimizeTestCase(TestCase):
+    def setUp(self):
+
+        self.cg_opt = {
+            'shrinker': 'frobenius_norm', 'verbose': 0, 'max_iter': 250,
+            'iter_callback': [], 'store_iterates': False,
+            'rel_tolerance': 1e-12, 'precision': 'float64',
+            'preconditioner': identity
+        }
+
+        rand_mat = rand((4, 4), seed=0)
+        self.A = rand_mat @ rand_mat.T
+        rand_mat_comp = rand((4, 4), seed=0) + rand((4, 4), seed=1)*1j
+        self.A_comp = rand_mat_comp @ rand_mat_comp.T.conjugate()
+
+        self.b = np.array(
+            [ 2.42430396e-09,  9.63863887e-10,  4.57019491e-10,  2.79965227e-10,
+              9.63863887e-10,  1.16254406e-09,  1.35936442e-09, -6.13130105e-10,
+              4.57019491e-10,  1.35936442e-09,  1.47323901e-09, -1.04452618e-09,
+              2.79965227e-10, -6.13130105e-10, -1.04452618e-09,  1.34539435e-09]
+        )
+
+        self.x = np.array(
+            [-1.59427196e-08, -2.78958759e-09,  2.00564628e-09,  3.64869830e-09,
+             -2.78958759e-09, -4.48456228e-09, -6.05281713e-09,  1.19913005e-08,
+              2.00564628e-09, -6.05281713e-09, -7.65604168e-09,  1.60532844e-08,
+              3.64869830e-09,  1.19913005e-08,  1.60532844e-08, -6.25894427e-09]
+        )
+
+        self.x_comp = np.array(
+            [ 4.08709701e-12+2.62480689e-11j,  1.62496340e-12+1.04358059e-11j,
+              7.70482177e-13+4.94817451e-12j,  4.71989099e-13+3.03119851e-12j,
+              1.62496340e-12+1.04358059e-11j,  1.95991527e-12+1.25869268e-11j,
+              2.29173171e-12+1.47179114e-11j, -1.03366668e-12-6.63839249e-12j,
+              7.70482177e-13+4.94817451e-12j,  2.29173171e-12+1.47179114e-11j,
+              2.48371114e-12+1.59508377e-11j, -1.76095073e-12-1.13091409e-11j,
+              4.71989099e-13+3.03119851e-12j, -1.03366668e-12-6.63839249e-12j,
+             -1.76095073e-12-1.13091409e-11j,  2.26817978e-12+1.45666567e-11j]
+        )
+
+    def tearDown(self):
+        pass
+
+    def testSolution(self):
+        self.assertTrue(np.allclose(apply(self.A, self.x), self.b)
+                        and np.allclose(apply(self.A_comp, self.x_comp), self.b*1j))
+
+    def testConjGradReal(self):
+        covar_coeff, _, _ = conj_grad(lambda x: apply(self.A, x), self.b, self.cg_opt)
+
+        self.assertTrue(np.allclose(covar_coeff, self.x, atol=1e-6))
```

Comment:
Why is this necessary?

Reply:
This was used to pass the old unit tests for Mac OS. It is  not necessary for new unit tests.